payments_service/braintree/forms.py

```diff
@@ -23,3 +31,41 @@ def clean(self):
         if method_missing or too_many_methods:
             raise forms.ValidationError(
                 'Either pay_method_nonce or pay_method_uri can be submitted')
+
+
+class ChangeSubscriptionPayMethodForm(forms.Form):
+    new_pay_method_uri = forms.CharField(max_length=255)
+    subscription_uri = forms.CharField(max_length=255)
+
+    def __init__(self, user, *args, **kw):
+        self.user = user
+        super(ChangeSubscriptionPayMethodForm, self).__init__(*args, **kw)
+
+    def clean_subscription_uri(self):
+        uri = self.cleaned_data['subscription_uri']
+        if not self.user_owns_resource(uri,
+                                       'paymethod__braintree_buyer__buyer'):
+            raise forms.ValidationError(
+                'subscription by URI does not exist or belongs to another user'
+            )
+        return uri
+
+    def clean_new_pay_method_uri(self):
+        uri = self.cleaned_data['new_pay_method_uri']
+        if not self.user_owns_resource(uri, 'braintree_buyer__buyer'):
+            raise forms.ValidationError(
+                'paymethod by URI does not exist or belongs to another user'
+            )
+        return uri
+
+    def user_owns_resource(self, uri, user_lookup_key):
+        try:
+            # Get the resource at the URI filtered by the logged in user.
+            # If the filtered result is empty (404) it means the user does not
+            # own the resource.
+            solitude.api().by_url(uri).get(**{user_lookup_key: self.user.pk})
+            return True
+        except HttpClientError, exc:
```

Comment:
Can't remember how much `HttpClientError` catches. If you did `get_object_or_404` you'd then be able to catch `ObjectNotFound` explicitly.